src/algorithms/libs/beidou_b1c_signal_processing.cc

```diff
@@ -0,0 +1,859 @@
+            else
+                {
+                    _dest[i] = std::complex<float>(1.0 - 2.0 * real_code_span[_codeValueIndex], 0);  //repeat the chip -> upsample
+                }
+        }           
+    volk_gnsssdr_free(real_code);
+}
+
+
+//--------------------------------------------------BOC_FOR_PILOT_COMPONENT----------------------------------------------------
+
+//! Generate BOC for first Pilot component which is in Real part
+void beidou_b1cp_gen_float_61(gsl::span<float> _dest, gsl::span<int> _prn)
+{
+	const uint32_t _codeLength = 12 * BEIDOU_B1Cp_CODE_LENGTH_CHIPS;
+	
+	const float alpha = sqrt(1.0 / 11.0);
+	
+	int32_t sinboc_61[12 * 10230] = {0};    //  _codeLength not accepted by Clang
+	gsl::span<int32_t> sinboc_61_(sinboc_61, _codeLength);
+    	   	
+   	beidou_b1c_pilot_sinboc_61_gen_int(sinboc_61_, _prn);  //generate sinboc(6,1) 12 samples per chip
+    		
+	for (uint32_t i = 0; i < _codeLength; i++)
+        {
+        	_dest[i] = alpha * static_cast<float>(sinboc_11[i]);
+                               
+	}
+}
+
+
+void beidou_b1cp_code_gen_float_sampled_boc_61(gsl::span<float> _dest, uint32_t _prn)
+{
+    // This function is based on the GNU software GPS for MATLAB in Kay Borre's book
+    auto _codeLength = static_cast<uint32_t>(BEIDOU_B1Cp_CODE_LENGTH_CHIPS);
+    auto* b1c_pilot_primary_code_chips = static_cast<int32_t*>(volk_gnsssdr_malloc(static_cast<uint32_t>(BEIDOU_B1Cp_CODE_LENGTH_CHIPS) * sizeof(int32_t), volk_gnsssdr_get_alignment()));
+    const int32_t _samplesPerChip =  12;
+
+    make_b1cp(gsl::span<int32_t>(b1c_pilot_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cp_CODE_LENGTH_CHIPS)), _prn); //generate Beidou B1C pilot code, 1 sample per chip
+
+    _codeLength = _samplesPerChip * BEIDOU_B1Cp_CODE_LENGTH_CHIPS;
+    std::unique_ptr<float> _signal_B1C_pilot{new float[_codeLength]};
+    gsl::span<float> _signal_B1C_pilot_span(_signal_B1C_pilot, _codeLength);
+
+    beidou_b1cp_gen_float_61(_signal_B1C_pilot_span, gsl::span<int>(b1c_pilot_primary_code_chips, static_cast<uint32_t>(BEIDOU_B1Cp_CODE_LENGTH_CHIPS)));  // generate cboc 12 samples per chip
+    
+    volk_gnsssdr_free(b1c_pilot_primary_code_chips);
+}
+
+
+//! Generate BOC for first Pilot component which is in Imaginary part
+void beidou_b1cp_gen_float_11(gsl::span<float> _dest, gsl::span<int> _prn)
+{
+	const uint32_t _codeLength = 12 * BEIDOU_B1Cp_CODE_LENGTH_CHIPS;
+	
+    	const float beta = sqrt(29.0 / 44.0);
+	
+	int32_t sinboc_11[12 * 10230] = {0};     //  _codeLength not accepted by Clang
+    	gsl::span<int32_t> sinboc_11_(sinboc_11, _codeLength);
+    	
+    	beidou_b1c_pilot_sinboc_11_gen_int(sinboc_11_, _prn);  //generate sinboc(1,1) 12 samples per chip
+    	
+	for (uint32_t i = 0; i < _codeLength; i++)
+        {
+        	_dest[i] = beta * static_cast<float>(sinboc_11[i]);
+                               
+	}
+}
+
+
+void beidou_b1cp_code_gen_float_sampled_boc_11(gsl::span<float> _dest, uint32_t _prn)
```

Comment:
The functions need to be added to .h file so that it can be accessed from external files. If you fix the acquisition comments you will notice this